// @chakra-ui/styled-system/dist/cjs/config/space.js
"use strict";

exports.__esModule = true;
exports.spaceParser = exports.space = void 0;

var _core = require("@styled-system/core");

var _utils = require("../utils");

var config = {
  margin: _utils.t.spaceT("margin"),
  marginTop: _utils.t.spaceT("marginTop"),
  marginBlockStart: _utils.t.spaceT("marginBlockStart"),
  marginRight: _utils.t.spaceT("marginRight"),
  marginInlineEnd: _utils.t.spaceT("marginInlineEnd"),
  marginBottom: _utils.t.spaceT("marginBottom"),
  marginBlockEnd: _utils.t.spaceT("marginBlockEnd"),
  marginLeft: _utils.t.spaceT("marginLeft"),
  marginInlineStart: _utils.t.spaceT("marginInlineStart"),
  marginX: _utils.t.spaceT(["marginLeft", "marginRight"]),
  marginInline: _utils.t.spaceT("marginInline"),
  marginY: _utils.t.spaceT(["marginTop", "marginBottom"]),
  marginBlock: _utils.t.spaceT("marginBlock"),
  padding: _utils.t.space("padding"),
  paddingTop: _utils.t.space("paddingTop"),
  paddingBlockStart: _utils.t.space("paddingBlockStart"),
  paddingRight: _utils.t.space("paddingRight"),
  paddingBottom: _utils.t.space("paddingBottom"),
  paddingBlockEnd: _utils.t.space("paddingBlockEnd"),
  paddingLeft: _utils.t.space("paddingLeft"),
  paddingInlineStart: _utils.t.space("paddingInlineStart"),
  paddingInlineEnd: _utils.t.space("paddingInlineEnd"),
  paddingX: _utils.t.space(["paddingLeft", "paddingRight"]),
  paddingInline: _utils.t.space("paddingInline"),
  paddingY: _utils.t.space(["paddingTop", "paddingBottom"]),
  paddingBlock: _utils.t.space("paddingBlock")
};
config.m = config.margin;
config.mt = config.marginTop;
config.mr = config.marginRight;
config.me = config.marginInlineEnd;
config.marginEnd = config.marginInlineEnd;
config.mb = config.marginBottom;
config.ml = config.marginLeft;
config.ms = config.marginInlineStart;
config.marginStart = config.marginInlineStart;
config.mx = config.marginX;
config.my = config.marginY;
config.p = config.padding;
config.pt = config.paddingTop;
config.py = config.paddingY;
config.px = config.paddingX;
config.pb = config.paddingBottom;
config.pl = config.paddingLeft;
config.ps = config.paddingInlineStart;
config.paddingStart = config.paddingInlineStart;
config.pr = config.paddingRight;
config.pe = config.paddingInlineEnd;
config.paddingEnd = config.paddingInlineEnd;
/**
 * Types for space related CSS properties
 */

/**
 * Converts shorthand or longhand margin and padding props to margin and padding CSS declarations
 *
 * - Numbers from 0-4 (or the length of theme.space) are converted to values on the spacing scale.
 * - Negative values can be used for negative margins.
 * - Numbers greater than the length of the theme.space array are converted to raw pixel values.
 * - String values are passed as raw CSS values.
 * - Array values are converted into responsive values.
 */
var space = (0, _core.system)(config);
exports.space = space;
var spaceParser = (0, _core.createParser)(config);
exports.spaceParser = spaceParser;
